src/api/app/models/cloud/azure/configuration.rb

```diff
@@ -0,0 +1,47 @@
+module Cloud
+  module Azure
+    class Configuration < ApplicationRecord
+      # maximum length has to be 245 (2048 bit RSA keys cannot encrypt more than 245 characters)
+      validates :application_id, presence: { message: 'ID can\'t be blank' },
+        length: { maximum: 245, too_long: 'ID is too long (maximum is 245 characters)' }, on: :update
+      validates :application_key, presence: true, length: { maximum: 245 }, on: :update
+
+      before_save :encrypt_credentials
+
+      belongs_to :user
+
+      def self.table_name_prefix
+        'cloud_azure_'
+      end
+
+      def available?
```

Comment:
Are you sure this works? As we have presence validations this should be always true (after the user visited the show page).

Reply:
@ChrisBr 

> Are you sure this works? As we have presence validations this should be always true (after the user visited the show page).

Tested again. `available?` is used when the user does not have any credential stored. You can see it in the images attached:

![azure_configuration_1](https://user-images.githubusercontent.com/24919/37395379-6693397c-2776-11e8-85bb-3cfb1e2578cf.png)

![azure_configuration_2](https://user-images.githubusercontent.com/24919/37395422-79af4f82-2776-11e8-983f-fa6f37f2f41d.png)